PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java

```diff
@@ -23,199 +24,219 @@
+                public void onClick(View v) {
+                    // Incase the user move back to map in between a running
+                    // Scenario.
+                    SessionHistory.totalPoints -= SessionHistory.currScenePoints;
+                    goToMap.setClickable(false);
+                    Intent myIntent = new Intent(GameActivity.this, MapActivity.class);
+                    startActivityForResult(myIntent, 0);
+                    getmDbHandler()
+                            .setReplayedScenario(scene.getScenarioName());
+                    goToMap.setAlpha((float) 0.0);
+                    replay.setAlpha((float) 0.0);
+                }
+            });
+            // Replay Mechanics
+            replay.setAlpha((float) 1.0);
+            replay.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    // In case the user moves back to map in between a running
+                    // Scenario.
+
+                    SessionHistory.totalPoints -= SessionHistory.currScenePoints;
+                    replay.setClickable(false);
+                    Intent myIntent = new Intent(GameActivity.this, GameActivity.class);
+                    startActivityForResult(myIntent, 0);
+                    getmDbHandler()
+                            .setReplayedScenario(scene.getScenarioName());
+                    goToMap.setAlpha((float) 0.0);
+                    replay.setAlpha((float) 0.0);
+                }
+            });
+        }
+        // If completed check if it is last scene
+        if (scene.getCompleted() == 1) {
+            if (scene.getNextScenarioID() == -1) {
+                Intent myIntent = new Intent(GameActivity.this, GameOverActivity.class);
+                finish();
+                startActivityForResult(myIntent, 0);
+            } else {
+                SessionHistory.currSessionID = scene.getNextScenarioID();
+                updateScenario();
+            }
+        }
+        SessionHistory.currQID = scene.getFirstQuestionID();
+        scenarioNameTextView.setText(scene.getScenarioName());
+        updateQA();
+    }
+
+    private void updateQA() {
+
+        listAdapter.clear();
+        getmDbHandler().getAllAnswer(answers, SessionHistory.currQID);
+        for (Answer ans : answers) {
+            listAdapter.add(ans.getAnswerDescription());
+        }
+        Question questions = getmDbHandler().getCurrentQuestion();
+        questionTextView.setText(questions.getQuestionDescription());
+    }
+
+    public DatabaseHandler getmDbHandler() {
+        return mDbHandler;
+    }
+
+    public void setmDbHandler(DatabaseHandler mDbHandler) {
+        this.mDbHandler = mDbHandler;
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == 1) {
```

Comment:
What does this 1 represent? Is there an enum holding this value you could reference instead?